### PR TITLE
[wasm] Fix perf pipeline

### LINKED ISSUE #87434: [wasm] Perf pipeline completely broken
`Zip Browser Wasm Artifacts` step failing with:

```
##[error]Error: Unable to locate executable file: 'zip'. Please verify either the file path exists or the file can be found within a directory specified by the PATH environment variable. Also check the file mode to verify the file is executable.
##[error]Unable to locate executable file: 'zip'. Please verify either the file path exists or the file can be found within a directory specified by the PATH environment variable. Also check the file mode to verify the file is executable.
```

Changes since last working build: 15d77612...22492751 which includes https://github.com/dotnet/runtime/pull/86806 .

### Known Issue Error Message

Fill the error message using [step by step known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssueJsonStepByStep.md#how-to-create-a-known-issue-step-by-step).

```json
{
  "ErrorMessage": "",
  "ErrorPattern": "",
  "BuildRetry": false,
  "ExcludeConsoleLog": false
}
```

#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|0|0|0|

### PR DESCRIPTION
**Pull request by radical**
- [wasm] Use the current `test-main.js` instead of the one for 7.0
- [wasm] perf: Add `--module` arg for v8, to support the es6 module
- [wasm] perf: Use tar for BrowserWasm for perf runs
- disable non-wasm builds

Fixes https://github.com/dotnet/runtime/issues/87434

### CODE REVIEW

**Comment by radical** [comment]
Waiting on the changes to get tested with `dotnet-runtime-perf` - `20230612.18`.

**Comment by radical** [comment]
> Waiting on the changes to get tested with `dotnet-runtime-perf` - `20230612.18`.

This passed. I want to merge this PR asap to unblock the perf pipeline.

File: eng/pipelines/coreclr/perf-wasm-prepare-artifacts-steps.yml [hunk 1]
```diff
@@ -16,7 +16,8 @@ steps:
       cp -r $(Build.SourcesDirectory)/artifacts/bin/dotnet-latest $(Build.SourcesDirectory)/artifacts/staging &&
       cp -r $(Build.SourcesDirectory)/artifacts/bin/microsoft.netcore.app.runtime.browser-wasm $(Build.SourcesDirectory)/artifacts/staging &&
       cp -r $(Build.SourcesDirectory)/artifacts/bin/microsoft.netcore.app.ref $(Build.SourcesDirectory)/artifacts/staging &&
-      cp -r $(Build.SourcesDirectory)/artifacts/packages/${{ parameters.configForBuild }}/Shipping/Microsoft.NET.Sdk.WebAssembly.Pack* $(Build.SourcesDirectory)/artifacts/staging/built-nugets
+      cp -r $(Build.SourcesDirectory)/artifacts/packages/${{ parameters.configForBuild }}/Shipping/Microsoft.NET.Sdk.WebAssembly.Pack* $(Build.SourcesDirectory)/artifacts/staging/built-nugets &&
+      find $(Build.SourcesDirectory)/artifacts/staging -type d
```
Comment (LoopedBard3) [inline eng/pipelines/coreclr/perf-wasm-prepare-artifacts-steps.yml:20]: Was the find just a testing change?

**Review by LoopedBard3 — APPROVED** [review]
One question, but otherwise looks good to me!

File: eng/pipelines/coreclr/perf-wasm-prepare-artifacts-steps.yml (hunk 1 above)
Comment (radical) [reply eng/pipelines/coreclr/perf-wasm-prepare-artifacts-steps.yml:20]: oops, yes!

**Comment by radical** [comment]
None of the changes here will affect the CI checks on the PR itself. They will get instead affect `dotnet-runtime-perf`. I'm merging this now to unblock the perf pipeline.

**Pull request merged**